Training24/NumConversion.c

```diff
@@ -0,0 +1,107 @@
+// ------------------------------------------------------------------------------------------------
+// Training ~ A training program for new joiners at Metamation, Batch - July 2024.
+// Copyright (c) Metamation India.
+// ------------------------------------------------------------------
+// Numconversion.c
+// Program to get user input(only integer) and convert them to Hexadecimal and Binary.
+// ------------------------------------------------------------------------------------------------
+#include <stdio.h>
+#include <stdbool.h>
+#include <string.h>
+#include <ctype.h>
+#include <limits.h>
+
+#define BITS_PER_HEX_DIGIT 4
+#define MAX_INPUT_LENGTH 100
+#define MAX_HEX_DIGITS 32
+
+/// <summary> Function to convert an integer to binary and return it as a string</summary>
+char* ConvertToBinary (int number, int bitWidth) {
+   static char binaryString[33];
+   for (int i = 0; i < bitWidth; i++) {
+      binaryString[i] = ((number >> (bitWidth - i - 1)) & 1) ? '1' : '0';
+   }
+   binaryString[bitWidth] = '\0';
+   return binaryString;
+}
+
+
+/// <summary>Function to  check if input is a valid integer.</summary>
+bool IsValidInteger (const char* str) {
+   if (*str == '-' || *str == '+') str++;   // Skip sign if present
+   if (!*str) return false;   // If string is empty after sign
+   while (*str) {   // Check each character to ensure it's a digit
+      if (!isdigit (*str)) return false; 
+      str++;
+   }
+   return true;
+}
+
+/// <summary>Function to get a valid integer from the user.</summary>
+bool GetValidInteger (int* number) {
+   char input[MAX_INPUT_LENGTH];
+   if (fgets (input, sizeof (input), stdin) != NULL) {

```

Comment:
```suggestion
   if (fgets (input, sizeof (input), stdin) != NULL) return false;
```
See [guard](https://en.wikipedia.org/wiki/Guard_(computer_science)).